shared/torngit/gitlab.py

```diff
@@ -440,6 +440,19 @@ async def list_repos(self, username=None, token=None):
         data = [i for n, i in enumerate(data) if i not in data[n + 1 :]]
         return data
 
+    async def list_repos_generator(self, username=None, token=None):
+        """
+        Unlike GitHub and Bitbucket, GitLab has to pull a complete list of repos
```

Comment:
Don't you think that if we keep some `set()` of yielded repos we can de-dup per group and behave more like the other implementations? It would also let us be more efficient, given that we can process a page while fetching the next group.

I don't think realistically that an extra set with repo ids will use that much memory. In fact it might even save some, given that we don't have to keep the data of all repos until the end.

Reply:
@giovanni-guidini i quickly tried that earlier but i didn't get tests passing, didn't investigate why

i think the population of onboarding gitlab users who would see the specific unimpressive (but not bugged) experience this change improves is quite small, and ultimately everybody (including them) is probably better served by the rest of the stuff on my plate. can always come back to this if we see signs that this is a common pain point